src/spec/expectations.cr

```diff
@@ -382,7 +382,7 @@ module Spec
     # If *message* is a regular expression, it is used to match the error message.
     #
     # It returns the rescued exception.
-    {% if flag?(:wasm32) %}
+    {% if flag?(:wasm32_mock_expect_raises) %}
```

Comment:
I don't understand this change. Why would you want to have the original expect_raises in wasm32?

Reply:
Otherwise the smoke tests fail to compile. This mock changes the return type (from the exception instance to nil) and a few of the spec files fail to compile. And if I don't mock, pretty much all tests fail to run.

Reply:
Ah, actually the real question is what's the need for smoke tests if this PR adds a proper test

Reply:
Ah, good question... I guess because it compiles all tests instead of ~half of them? It will get less and less useful as time goes by. 

Shall I remove the smoke tests?

Reply:
Yes, when adding *actually* running test we can remove the compile-only smoke tests. Even when they only have partial coverage. It's certainly better than anything a smoke test could provide.

Reply:
(Arguably, there's still a low chance to miss changes affecting currently untested code, which a smoke test would catch. But more and more running specs will be enabled in the future at which point such defects will be detected and fixed.)

Reply:
Minor comment: I'd say the chances are not that low. Yet, in a vast majority the error will be fixed raising a "not implemented", which isn't a big difference. So yeah, test > smoke.

Reply:
Yeah, I meant a low chance to miss a relevant change. One that accidentally breaks something that used to work before and would be discovered in the smoke test.